### PR TITLE
View accessibility invoke

### PR DESCRIPTION
**Pull request by ahimberg**
This is some progress on improved accessibility for Touchables so they can describe themselves as a button to accessibility tools and be invokable.  This is based on the C# support

In this change I've modified ViewViewManager to use a custom ContentControl (called ViewControl) which then has a custom automationpeer that will look at the accessibleRole to decide if it should expose itself as a button.

To build the custom automation peer I needed to use the cppwinrt codegen tools to generate the templates so we could override some methods.  For this change I'm submitting the generated files and will be following up with getting codegen of that in the build system - its a painful to do that in obuild and cmake.  Also you'll see some small ifdefs to support obuild where we are not on the final RS5 SDK so  needed to condition some stuff to get things to build there.

Next up we need to hide the children so we don't get text inside of button in the acc tree, and support concat'ing all the text children when accessibilityLabel hasn't been set.

related issues:
https://github.com/Microsoft/react-native-windows/issues/2122
https://github.com/Microsoft/react-native-windows/issues/2089

### CODE REVIEW

**Comment by YuliKl** [comment]
This comment

> Next up we need to hide the children so we don't get text inside of button in the acc tree

feels directly related to #2161 that I had opened.

File: RNWCPP/ReactUWP/Views/ViewControl.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+#pragma once
+
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.UI.Xaml.h>
+#include <winrt/Windows.UI.Xaml.Automation.Peers.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
+#include <winrt/Windows.UI.Xaml.Media.h>
+
+#include <Views/ShadowNodeBase.h>
```
Comment (randy-flynn) [inline RNWCPP/ReactUWP/Views/ViewControl.h]: This should go above all the winrt includes.

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp [hunk 2]
```diff
@@ -225,6 +247,22 @@ XamlView ViewViewManager::CreateViewControl(int64_t tag)
...
+  contentControlPtr->AccessibilityInvoke([=]()
+  {
+    auto instance = m_wkReactInstance.lock();
+    if (instance != nullptr)
+    {
+      auto pNativeUiManager = static_cast<NativeUIManager*>(instance->NativeUIManager());
+      facebook::react::INativeUIManagerHost* pUIManagerHost = pNativeUiManager->getHost();
+
+      auto pViewShadowNode = static_cast<ViewShadowNode*>(pUIManagerHost->FindShadowNodeForTag(tag));
+      if (pViewShadowNode != nullptr && pViewShadowNode->OnClick())
```
Comment (randy-flynn) [inline RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: I know the ShadowNode should be there, but do we really want to hit the else statement if it's not?

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp [hunk 3]
```diff
@@ -306,8 +346,47 @@ void ViewViewManager::UpdateProperties(ShadowNodeBase* nodeToUpdate, folly::dyna
...
+          shouldBeControl = propertyValue.asBool();
+      }
+      else if (propertyName == "accessibilityRole")
+      {
+        // FUTURE having an accessibilityRole shouldBeControl to support
+        // non-Touchable scenarios
+        if (propertyValue.isString())
+        {
+          auto role = propertyValue.asString();
+          if (role == "none")
```
Comment (randy-flynn) [inline RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: If the role isn't one we support then this code will leave it at the previous value. Is that the right thing to do or should we set it to None?

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp (hunk 2 above)
Comment (ahimberg) [reply RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: for the other events we don't check that, all we need is a tag

File: RNWCPP/ReactUWP/Views/ViewViewManager.cpp (hunk 3 above)
Comment (ahimberg) [reply RNWCPP/ReactUWP/Views/ViewViewManager.cpp]: i'll add else do none

File: RNWCPP/ReactUWP/Views/ViewControl.h (hunk 1 above)
Comment (ahimberg) [reply RNWCPP/ReactUWP/Views/ViewControl.h]: next rev

**Pull request merged**